crypto/crypto-derive/src/hasher.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Converts a camel-case string to snake-case
+pub fn camel_to_snake(text: &str) -> String {
+    let mut out = String::with_capacity(text.len());
+    let mut first = true;
+    text.chars().for_each(|c| {
+        if !first && c.is_uppercase() {
+            out.push('_');
+            c.to_lowercase().for_each(|c| out.push(c));
```

Comment:
This method presents two ways for pushing lowercase chars. Let's be consistent and use 
```
let c_low = c.to_lowercase().next().unwrap();
out.push(c_low);
```

Reply:
Fixed!